### PR TITLE
adding Secret Santa sample

### PR DESCRIPTION
**Pull request by delbert**
hello Everyone, i would like to add the Secret Santa sample to these samples, please.  i have just finished the Jupyter Notebook which i hope explains the sample well.

### CODE REVIEW

File: samples/secret-santa/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+author: delbert murphy
+ms.author: delbertm@microsoft.com
+products:
+- azure-qio
+- azure-quantum
+description: "Learn how to solve a binary optimization problem using the Azure Quantum optimization service"
+urlFragment: azure-quantum.secret-santa
+---
+
+# Solving the secret santa scenario with the Azure Quantum optimization service
```
Comment (anraman) [inline samples/secret-santa/README.md:14]: Santa should be capitalised, would suggest just capitalising Secret Santa everywhere for simplicity

```suggestion
# Solving the Secret Santa scenario with the Azure Quantum optimization service
```

File: samples/secret-santa/secret_santa.ipynb [hunk 2]
```diff
@@ -0,0 +1,314 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d25099d2",
+   "metadata": {},
+   "source": [
+    "Copyright (c) Microsoft Corporation.\n",
```
Comment (anraman) [inline samples/secret-santa/secret_santa.ipynb:8]: We don't need the MIT license bit at the start of the notebook, there's a licence bit included in the main repo

**Review by anraman — CHANGES_REQUESTED** [review]
Partial review, still working through the notebook. Overall, sample looks good - mostly cosmetic updates needed. Please could you modify all files submitted to use single spaces following a period to bring this sample in line with the others, style-wise?

File: samples/secret-santa/secret-santa.ipynb [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    "\r\n",
+    "To represent the Secret Santa problem, we can use six binary variables, as outlined in the Scenario Table below:\r\n",
+    "\r\n",
+    "|- buys ->|**Vincent**|**Tess**|**Uma**|\r\n",
+    "|--|--|--|--|\r\n",
+    "|**Vincent**|--|$x_0$|$x_1$|\r\n",
+    "|**Tess**|$x_2$|--|$x_3$|\r\n",
+    "|**Uma**|$x_3$|$x_4$|--|\r\n",
+    "\r\n",
+    "The constraints for the problem can be expressed as doing logical ANDs ($ \\land $) of variables that are EXLUSIVE-ORd ($ \\oplus $) together, like this:\r\n",
```
Comment (frtibble) [inline samples/secret-santa/secret-santa.ipynb:58]: NIT: typo "exclusive"

File: samples/secret-santa/secret-santa.ipynb [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+   "source": [
+    "So, how do we represent $ ( x_0 \\oplus x_1 ) $ in a binary format that Azure Quantum will understand?\r\n",
+    "\r\n",
+    "Keeping in mind that we want to **minimize** our cost function, let's try to use the following representation:\r\n",
+    "\r\n",
+    "$ ( x_0 + x_1 - 1 )^2 $  \r\n",
+    "\r\n",
+    "Let's check the truth table for this formulation:\r\n",
+    "\r\n",
+    "|$x_0$|$x_1$|$(x_0 + x_1 - 1)^2$|\r\n",
```
Comment (frtibble) [inline samples/secret-santa/secret-santa.ipynb:106]: A suggestion: like in the shipping sample, we have a follow-up section on how to simplify the problem formulation, I think we could have a similar additional section which could show how to phrase this same problem using `SLCTerms`. I don't suggest doing it in this PR, just mostly making a note...

Comment (frtibble) [reply samples/secret-santa/secret-santa.ipynb:106]: Just got to the note at the end that suggests this!

File: samples/secret-santa/secret-santa.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    terms.append(Term(c = 1.0, indices = [i, i]))   # x(i)^2
+    terms.append(Term(c = 1.0, indices = [j, j]))   # x(j)^2
+    terms.append(Term(c = 2.0, indices = [i, j]))   # 2x(i)x(j) 
+    terms.append(Term(c = -2.0, indices = [i]))     # -2x(i)
+    terms.append(Term(c = -2.0, indices = [j]))     # -2x(j)
+    terms.append(Term(c = 1.0, indices = []))       # +1
+
+    return terms
+
+# Helper funciton to interpret the answer returned by the service in a human-readable way:
```
Comment (frtibble) [inline samples/secret-santa/secret-santa.py:40]: NIT: typo "function"

**Review by frtibble — APPROVED** [review]

File: samples/secret-santa/secret-santa.ipynb (hunk 3 above)
Comment (anraman) [reply samples/secret-santa/secret-santa.ipynb:58]:
```suggestion
    "The constraints for the problem can be expressed as doing logical ANDs ($ \\land $) of variables that are EXCLUSIVE-ORd ($ \\oplus $) together, like this:\r\n",
```

File: samples/secret-santa/secret-santa.py (hunk 5 above)
Comment (anraman) [reply samples/secret-santa/secret-santa.py:40]:
```suggestion
# Helper function to interpret the answer returned by the service in a human-readable way:
```

**Review by anraman — CHANGES_REQUESTED** [review]

**Review by anraman — APPROVED** [review]

**Pull request merged**